### ISSUE TITLE
[New build - DEFAULT] 23w38a: Solar Panel, Helmet, Gas Bottle

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#5287]
This build adds a few components needed for Survival mode.

Backpack:
The player can now store a few items in its backpack on top of the hotbar. 
Final quantity will tweaked.
There is a new tab in the codex, and a new backpack transfer window when opening an inventory block.

Solar Panels
In the early game the solar panel will help the player to produce power as hydrogen will not be available.
Solar panels produces a various amount of power depending on their incidence to the sun.
They are not checking for shadow coverage yet.

Helmet:
It is now possible to equip an helmet! Simply press H by default to swap it.

Gas Bottle:
The gas bottle can be used to manually transfer fuel or gas from one tank to another.
Be careful to not hit them as they might gets off with a bang!

Also includes various fixes and improvements related to crafting and tank.

**Comment by TIKIRobo** [comment]
quick question, now that we have solar panels will we get batteries? the solar panels do store energy but i don't think they store enough for most craft

**Comment by TIKIRobo** [comment]
small bug, solar panels don't really seem to work, i grabbed one of ProPeach's tie fighters off the ws and tried adding the solar panels to the wings, after removing all of the ships reactors i now have a power balance of -166% 
![image](https://github.com/tsunamayo/Starship-EVO/assets/56371294/904eb131-e55b-4927-b136-ae5ccab7571d)


[TIE-IN but real.zip](https://github.com/tsunamayo/Starship-EVO/files/12705465/TIE-IN.but.real.zip)

**Comment by BigBadKangaroo** [comment]
Hi there,

I really like the new helmet, but it looks a bit off imo ... so I made a small photoshop version of how I think it should look like:
Before:
![20230923105314_1](https://github.com/tsunamayo/Starship-EVO/assets/81495956/758c9c97-2187-4c3f-baa2-38117c0d1235)

After: (sorry for my bad PS skills lol)
![20230923105314_1X](https://github.com/tsunamayo/Starship-EVO/assets/81495956/586eb882-9a92-43d9-b740-603d57d3035e)

I know, it is so large that every skin can have the same helmet, but I think there aren't too many skins at all and maybe it would be a good idea to give them each a helmet size that suits the individual skin well instead of one universal helmet size

Just my thoughts on that

**Comment by ProPeach** [comment]
What a surprise! I wasn't expecting solar panels for a long while, they look great. The gas bottles are super fun too, will the tank bricks also burn and explode if shot too?

**Comment by Kaiser-Indrasil** [comment]
Hey, what fixes and tweaks are you talking about Tsuna?

**Comment by tsunamayo** [comment]
@BigBadKangaroo yep I will see what I can do. @ProPeach in time I want to also add that!
@Kaiser-Indrasil well I havent kept track of all of them as I also did a bunch of refactoring. It was mostly linked to furnace / assembler not pulling items, or pushing them to the wrong inventory.

**Comment by tsunamayo** [comment]
@TIKIRobo Thanks I will check that issue!

**Comment by Fam-of3** [comment]
When selecting the solar panels in build mode, the UI text readout on the right of the screen states the held object is <sprite=-1> for both square and angled variants.
![StarshipEVO_BUGS_UI_solarpanels_00](https://github.com/tsunamayo/Starship-EVO/assets/19860341/6cb01f02-a15f-4318-9ca9-b0e4fed8adf3)

**Comment by tsunamayo** [comment]
@Fam-of3 @TIKIRobo Issues are fixed in new 23w39a. Thanks
